Plugins/SpatialGDK/SpatialGDKEditorToolbar/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorGenerateSnapshot.cpp

```diff
@@ -120,52 +129,181 @@ bool CreateGlobalStateManager(Worker_SnapshotOutputStream* OutputStream)
+void SetupStartupActorCreation(USpatialNetDriver*& NetDriver, USpatialNetConnection*& NetConnection, USpatialTypebindingManager*& TypebindingManager)
 {
-	// Set up grid of "placeholder" entities to allow workers to be authoritative over _something_.
-	int PlaceholderCount = SpatialConstants::PLACEHOLDER_ENTITY_ID_LAST - SpatialConstants::PLACEHOLDER_ENTITY_ID_FIRST + 1;
-	int PlaceholderCountAxis = static_cast<int>(sqrt(PlaceholderCount));
-	checkf(PlaceholderCountAxis * PlaceholderCountAxis == PlaceholderCount, TEXT("The number of placeholders must be a square number."));
-	checkf(PlaceholderCountAxis % 2 == 0, TEXT("The number of placeholders on each axis must be even."));
-	const float CHUNK_SIZE = 5.0f; // in SpatialOS coordinates.
-	int PlaceholderEntityIdCounter = SpatialConstants::PLACEHOLDER_ENTITY_ID_FIRST;
-	for (int x = -PlaceholderCountAxis / 2; x < PlaceholderCountAxis / 2; x++)
+	TypebindingManager = NewObject<USpatialTypebindingManager>();
+	TypebindingManager->Init();
+
+	NetDriver = NewObject<USpatialNetDriver>();
+	NetDriver->ChannelClasses[CHTYPE_Actor] = USpatialActorChannel::StaticClass();
+	NetDriver->TypebindingManager = TypebindingManager;
+
+	NetConnection = NewObject<USpatialNetConnection>();
+	NetConnection->Driver = NetDriver;
+	NetConnection->State = USOCK_Closed;
+}
+
+void CleanupNetDriverAndConnection(USpatialNetDriver* NetDriver, USpatialNetConnection* NetConnection)
+{
+	// On clean up of the NetDriver due to garbage collection, either the ServerConnection or ClientConnections need to be not nullptr.
+	// However if the ServerConnection is set on creation, using the FObjectReplicator to create the initial state of the actor,
+	// the editor will crash. Therefore we set the ServerConnection after we are done using the NetDriver.
+	NetDriver->ServerConnection = NetConnection;
+}
+
+TArray<Worker_ComponentData> CreateStartupActorData(USpatialActorChannel* Channel, AActor* Actor, USpatialTypebindingManager* TypebindingManager, USpatialNetDriver* NetDriver)
+{
+	FClassInfo* Info = TypebindingManager->FindClassInfoByClass(Actor->GetClass());
+	check(Info);
+
+	FRepChangeState InitialRepChanges = Channel->CreateInitialRepChangeState(Actor);
+	FHandoverChangeState InitialHandoverChanges = Channel->CreateInitialHandoverChangeState(Info);
+
+	// Created just to satisfy the ComponentFactory constructor
+	FUnresolvedObjectsMap UnresolvedObjectsMap;
+	FUnresolvedObjectsMap HandoverUnresolvedObjectsMap;
+	ComponentFactory DataFactory(UnresolvedObjectsMap, HandoverUnresolvedObjectsMap, NetDriver);
+
+	// Create initial state of Actor (which is the state the Actor is in before running the level)
+	TArray<Worker_ComponentData> ComponentData = DataFactory.CreateComponentDatas(Actor, InitialRepChanges, InitialHandoverChanges);
+
+	// Actor RPCs
+	for (int RPCType = 0; RPCType < RPC_Count; RPCType++)
+	{
+		ComponentData.Add(ComponentFactory::CreateEmptyComponentData(Info->RPCComponents[RPCType]));
+	}
+
+	// Process Subobjects
```

Comment:
Wherever possible, let's avoid non-helpful comments and give more color to what's going on. "Process" is usually a bad go-to word in this sense.